test/controllers/api/v0/imports_controller_test.rb

```diff
@@ -19,24 +19,8 @@ def diff(_change, _options = nil)
     end
   end
 
-  # These tests get network privileges (for now)
-  def setup
-    WebMock.allow_net_connect!
-    @original_allowed_hosts = Archiver.allowed_hosts
-    Archiver.allowed_hosts = ['https://test-bucket.s3.amazonaws.com']
-    # Imports trigger analysis, which uses these diffs
-    Differ.register('html_source_dmp', MockDiffer.new)
-    Differ.register('html_text_dmp', MockDiffer.new)
-    Differ.register('links_json', MockDiffer.new)
-  end
-
-  def teardown
-    WebMock.disable_net_connect!
-    Archiver.allowed_hosts = @original_allowed_hosts
-  end
-
-  test 'can import data' do
-    import_data = [
+  def import_data
```

Comment:
Is there a reason you extracted this out, even though it’s only used in one function? All the other test functions still define their own `import_data` internally, since it’s mostly unique in every one.

Reply:
I'll inline it. I am struggling a bit with the mintest `test_` style because it seems like each one requires a lot of duplicated setup code. 

If you're ok with it, in future work I might explore minitest's spec syntax (`describe`, `it`) because it makes it a bit easier to create shared context across multiple tests.

Reply:
👍

Reply:
I am cool with that. I also have no objection if you want to migrate things over to RSpec, too.

IIRC, when I first started, I didn’t know Minitest had that syntax, and later when I discovered it, a bunch of things actually worked in a significantly different way (and I had trouble finding good resources on it), so I didn’t feel like it was worth the effort at the time. (Sadly, I never wrote any of that down and don’t remember details, so take that however you will.) And the use of Minitest in general was just because I didn’t think this project or the tests would get that big, and RSpec seemed like it might be an unnecessary large dependency. Obviously that hasn’t really held true.